pkg/controllers/sync/accessor.go

```diff
@@ -22,50 +21,50 @@ are Copyright 2023 The KubeAdmiral Authors.
 package sync
 
 import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
 	pkgruntime "k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/klog/v2"
 
 	fedcorev1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
-	genericclient "github.com/kubewharf/kubeadmiral/pkg/client/generic"
+	fedcorev1a1client "github.com/kubewharf/kubeadmiral/pkg/client/clientset/versioned/typed/core/v1alpha1"
+	fedcorev1a1informers "github.com/kubewharf/kubeadmiral/pkg/client/informers/externalversions/core/v1alpha1"
 	"github.com/kubewharf/kubeadmiral/pkg/controllers/common"
 	"github.com/kubewharf/kubeadmiral/pkg/controllers/sync/version"
-	"github.com/kubewharf/kubeadmiral/pkg/controllers/util"
+	"github.com/kubewharf/kubeadmiral/pkg/util/eventhandlers"
+	"github.com/kubewharf/kubeadmiral/pkg/util/fedobjectadapters"
+	"github.com/kubewharf/kubeadmiral/pkg/util/informermanager"
 )
 
 // FederatedResourceAccessor provides a way to retrieve and visit
 // logical federated resources (e.g. FederatedConfigMap)
 type FederatedResourceAccessor interface {
-	Run(stopChan <-chan struct{})
+	Run(context.Context)
 	HasSynced() bool
 	FederatedResource(
 		qualifiedName common.QualifiedName,
-	) (federatedResource FederatedResource, possibleOrphan bool, err error)
-	VisitFederatedResources(visitFunc func(obj interface{}))
+	) (federatedResource FederatedResource, err error)
+	VisitFederatedResources(visitFunc func(fedcorev1a1.GenericFederatedObject))
```

Comment:
Can we extract this method into a separate helper function or a method of the sync controller? I feel that it doesn't really make sense here.

Reply:
The comment for `FederatedResourceAccessor` states it "provides a way to retrieve and visit" `FederatedResource`s. It's basically a wrapper for the FO and CFO informers. The sync controller doesn't have the informer reference to implement this method.